pkg/arvo/app/group-hook.hoon

```diff
@@ -0,0 +1,270 @@
+/-  *group-store, *group-hook
```

Comment:
A brief comment wold be useful.  In particular, it's not immediately obvious what "synced" means.  My first guess would have been that it's a local copy of a foreign group, but it looks like actually it means that a foreign group has been used as input for a local group.  Thus if `~bud` is "synced" to `~dev`'s `/hello` group, then `~bud`'s `/hello` group will be a superset of `~dev`'s.  Is that correct?

Reply:
If ~bud adds ~dev/help to his synced groups, then ~dev/help will remove ~bud/help and write the contents of ~dev/help on top of the empty ~bud/help.

Reply:
Ah, so there's effectively one global namespace and you can choose to use your own version of a particular group path or someone else's.

Reply:
> A brief comment wold be useful. 

Agreed all of the files in this PR could use a short explanation of what they do at the top of the file.

Reply:
Added comments to all of the actions in sur files and to the top of the app files.